src/main/java/net/minestom/server/event/handler/EventHandler.java

```diff
@@ -109,22 +109,27 @@
      */
     default <E extends Event> void callEvent(@NotNull Class<E> eventClass, @NotNull E event) {
 
-        // Global listeners
-        if (!(this instanceof GlobalEventHandler)) {
-            final GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
-            runEvent(globalEventHandler.getEventCallbacks(eventClass), event);
-        }
+        try {
+
+            // Global listeners
+            if (!(this instanceof GlobalEventHandler)) {
+                final GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
+                runEvent(globalEventHandler.getEventCallbacks(eventClass), event);
+            }
 
-        // Local listeners
-        final Collection<EventCallback> eventCallbacks = getEventCallbacks(eventClass);
-        runEvent(eventCallbacks, event);
+            // Local listeners
+            final Collection<EventCallback> eventCallbacks = getEventCallbacks(eventClass);
+            runEvent(eventCallbacks, event);
 
-        // Call the same event for the current entity instance
-        if (this instanceof Entity) {
-            final Instance instance = ((Entity) this).getInstance();
-            if (instance != null) {
-                runEvent(instance.getEventCallbacks(eventClass), event);
+            // Call the same event for the current entity instance
+            if (this instanceof Entity) {
+                final Instance instance = ((Entity) this).getInstance();
+                if (instance != null) {
+                    runEvent(instance.getEventCallbacks(eventClass), event);
+                }
             }
+        } catch (Exception exception) {
+            MinecraftServer.getExceptionManager().handleException(exception);
```

Comment:
Chould this be done per-event callback instead? If one callback throws, the remaining callbacks won't be called right?

Reply:
Yeah, most likely